pkgs/applications/misc/xmrig/xmrig-cuda.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv
+, autoAddOpenGLRunpathHook
+, fetchFromGitHub
+, cmake
+, pkgs
+, tree
+, cudaArch ? 61
+}:
+
+let
+  libcudaDir = "/run/opengl-driver/lib";
+in
+stdenv.mkDerivation rec {
+  inherit (pkgs.cudaPackages) cudatoolkit cuda_cudart;
+
+  version = "6.17.0";
+  pname = "xmrig-cuda";
+  src = fetchFromGitHub {
+    owner = "xmrig";
+    repo = "xmrig-cuda";
+    rev = "v${version}";
+    sha256 = "sha256-vRVLmMgCG42btm0jcteZiftchjRZotBgT98znQV0I+k=";
+  };
+
+  nativeBuildInputs = [ cmake autoAddOpenGLRunpathHook ];
+
+  buildInputs = [
+    cudatoolkit
+    cuda_cudart
+  ];
+
+  cmakeFlags = [
+    "-DCUDA_TOOLKIT_ROOT_DIR=${cudatoolkit}"
+    "-DLIBCUDA_LIBRARY_DIR=${libcudaDir}"
+    "-DCUDA_LIB=${libcudaDir}/libcuda.so"
```

Comment:
Ok, so upstream is trying to locate `libcuda.so` at build time. They really shouldn't be doing that, but since they do, we can temporarily give them a fake `libcuda.so`, a so called "stub library":

```
    "-DCUDA_LIB=${cuda_cudart}/lib/stubs/libcuda.so"
```

Once the build is finished, nix will prepend `/run/opengl-driver/lib` to the libraries' Runpaths, so that `/run/opengl-driver/lib` will take priority over the stub.